x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form_fields.tsx

```diff
@@ -113,30 +124,37 @@ export const PipelineFormFields: React.FunctionComponent<Props> = ({
       </FormRow>
 
       {/* Pipeline Processors Editor */}
-      <FormDataProvider pathsToWatch={['processors', 'on_failure']}>
-        {({ processors, on_failure: onFailure }) => {
-          const processorProp =
-            typeof processors === 'string' && processors
-              ? JSON.parse(processors)
-              : initialProcessors ?? [];
-
-          const onFailureProp =
-            typeof onFailure === 'string' && onFailure
-              ? JSON.parse(onFailure)
-              : initialOnFailureProcessors ?? [];
 
-          return (
-            <PipelineProcessorsEditor
-              onFlyoutOpen={onEditorFlyoutOpen}
-              esDocsBasePath={services.documentation.getEsDocsBasePath()}
-              isTestButtonDisabled={isTestButtonDisabled}
-              onTestPipelineClick={onTestPipelineClick}
-              onUpdate={onProcessorsUpdate}
-              value={{ processors: processorProp, onFailure: onFailureProp }}
-            />
-          );
-        }}
-      </FormDataProvider>
+      <PipelineProcessorsContextProvider
+        onFlyoutOpen={onEditorFlyoutOpen}
+        links={{ esDocsBasePath: services.documentation.getEsDocsBasePath() }}
+        onUpdate={onProcessorsUpdate}
+        value={{ processors, onFailure }}
+      >
+        <div className="pipelineProcessorsEditor">
+          <EuiFlexGroup gutterSize="m" responsive={false} direction="column">
+            <EuiFlexItem grow={false}>
+              <ProcessorsTitle
```

Comment:
nit: I think I would have preferred `<ProcessorsHeader>` as a name. As it contains the title but also buttons.